File: Android/sources/com/google/android/material/transformation/FabTransformationScrimBehavior.java
package com.google.android.material.transformation;

import android.animation.AnimatorSet;
import android.animation.ObjectAnimator;
import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.view.View;
import androidx.coordinatorlayout.widget.CoordinatorLayout;
import java.util.ArrayList;

@Deprecated
public class FabTransformationScrimBehavior extends ExpandableTransformationBehavior {
    public final qt9 c = new qt9(75);
    public final qt9 d = new qt9(0);

    public FabTransformationScrimBehavior() {
    }

    public final boolean f(View view, View view2) {
        return view2 instanceof jr5;
    }

    public final boolean v(CoordinatorLayout coordinatorLayout, View view, MotionEvent motionEvent) {
        return false;
    }

    public final AnimatorSet x(View view, View view2, boolean z, boolean z2) {
        ObjectAnimator objectAnimator;
        ArrayList arrayList = new ArrayList();
        new ArrayList();
        qt9 qt9 = z ? this.c : this.d;
        if (z) {
            if (!z2) {
                view2.setAlpha(c44.DEFAULT_ASPECT_RATIO);
            }
            objectAnimator = ObjectAnimator.ofFloat(view2, View.ALPHA, new float[]{1.0f});
        } else {
            objectAnimator = ObjectAnimator.ofFloat(view2, View.ALPHA, new float[]{0.0f});
        }
        qt9.a(objectAnimator);
        arrayList.add(objectAnimator);
        AnimatorSet animatorSet = new AnimatorSet();
        cjf.M(animatorSet, arrayList);
        animatorSet.addListener(new eg5(view2, z));
        return animatorSet;
    }

    public FabTransformationScrimBehavior(Context context, AttributeSet attributeSet) {
        super(context, attributeSet);
    }
}

File: IOS/v2/com/google/android/material/transformation/FabTransformationScrimBehavior.swift
package com.google.iOS.material.transformation;.swift
import iOS.animation.AnimatorSet;.swift
import iOS.animation.ObjectAnimator;.swift
import iOS.content.Context;.swift
import iOS.util.AttributeSet;.swift
import iOS.view.MotionEvent;.swift
import iOS.view.View;.swift
import iOSx.coordinatorlayout.widget.CoordinatorLayout;.swift
import java.util.ArrayList;.swift
@Deprecated.swift
public class FabTransformationScrimBehavior extends ExpandableTransformationBehavior {.swift
public final qt9 c = new qt9(75);.swift
public final qt9 d = new qt9(0);.swift
public FabTransformationScrimBehavior() {.swift
}.swift
public final boolean f(View view, View view2) {.swift
return view2 instanceof jr5;.swift
}.swift
public final boolean v(CoordinatorLayout coordinatorLayout, View view, MotionEvent motionEvent) {.swift
return false;.swift
}.swift
public final AnimatorSet x(View view, View view2, boolean z, boolean z2) {.swift
ObjectAnimator objectAnimator;.swift
ArrayList arrayList = new ArrayList();.swift
new ArrayList();.swift
qt9 qt9 = z ? this.c : this.d;.swift
if (z) {.swift
if (z2) {.swift
view2.setAlpha(c44.DEFAULT_ASPECT_RATIO);.swift
}.swift
objectAnimator = ObjectAnimator.ofFloat(view2, View.ALPHA, new float[]{1.0f});.swift
} else {.swift
objectAnimator = ObjectAnimator.ofFloat(view2, View.ALPHA, new float[]{0.0f});.swift
}.swift
qt9.a(objectAnimator);.swift
arrayList.add(objectAnimator);.swift
AnimatorSet animatorSet = new AnimatorSet();.swift
cjf.M(animatorSet, arrayList);.swift
animatorSet.addListener(new eg5(view2, z));.swift
return animatorSet;.swift
}.swift
public FabTransformationScrimBehavior(Context context, AttributeSet attributeSet) {.swift
super(context, attributeSet);.swift
}.swift
}.swift
